core/database/src/main/java/com/google/samples/apps/nowinandroid/core/database/model/NewsResourceFtsEntity.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.core.database.model
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.Fts4
+
+/**
+ * Fts entity for the news resources. See https://developer.android.com/reference/androidx/room/Fts4.
+ */
+@Entity(tableName = "newsResourcesFts")
+@Fts4
```

Comment:
Is the new entity and DAO required? Can't we mark the old one as FTS?

Reply:
It should be separated because the columns in a fts table will be part of the searched contents by the `WHERE ftsTable MATCH :query`. We don't want to search against some columns, for example URL of the resource.